### PR TITLE
Support unix and bare sockets

### LINKED ISSUE #42: Support unix sockets in http2-client-grpc
The [`GrpcClientConfig`](https://github.com/haskell-grpc-native/http2-grpc-haskell/blob/496e92bc967eff02ac3698ba12ba2dfe38bc8b74/http2-client-grpc/src/Network/GRPC/Client/Helpers.hs#L65-L86) only allows connection by `Hostname` and `PortNumber`. It would be nice if it somehow supported Unix sockets too. Additionally, an option to take `Network.Socket` would be nice too.

### PR DESCRIPTION
**Pull request by akshaymankar**
Fixes #42 

I am not entirely certain about using `localhost` as the Authority. I added support for bare sockets because it seemed easy enough.

### CODE REVIEW

File: http2-client-grpc/src/Network/GRPC/Client/Helpers.hs [hunk 1]
```diff
@@ -61,12 +63,14 @@ data BackgroundTasks = BackgroundTasks {
   -- ^ Periodically ping the server.
   }
 
+data Address = AddressTCP HostName PortNumber
```
Comment (lucasdicioccio) [inline http2-client-grpc/src/Network/GRPC/Client/Helpers.hs:66]: +add some comment along the lines of "a generalized address that supports, new TCP connections, new UNIX socket creations, or using already-connected sockets"

**Review by lucasdicioccio — COMMENTED** [review]
The code aspects of the patch looks good to me.

Can you:
- grep to see if some examples / the bench tool (i believe nothing does) needs updating
- add an entry to the changelog (credit yourself for the patch <3 )

**Comment by akshaymankar** [comment]
> * grep to see if some examples / the bench tool (i believe nothing does) needs updating

I couldn't find anything. I compiled all the packages, nothing failed.
 
> * add an entry to the changelog (credit yourself for the patch <3 )

I added a link to this PR.

**Comment by lucasdicioccio** [comment]
Thank you Akshay!

The build before the doc+changelog commits worked, it looks like it timed out this one time. Restarting it would likely mostly waste resources at Travis => merging that.

**Pull request merged**